tests/parser/features/external_contracts/test_external_contract_calls.py

```diff
@@ -795,3 +740,87 @@ def get_array(arg1: address) -> int128[3]:
 
     c2 = get_contract(contract_2)
     assert c2.get_array(c.address) == [0, 0, 0]
+
+
+def test_returndatasize_too_short(get_contract, assert_tx_failed):
+    contract_1 = """
+@external
+def bar(a: int128) -> int128:
+    return a
+"""
+    contract_2 = """
+interface Bar:
+    def bar(a: int128) -> (int128, int128): view
+
+@external
+def foo(_addr: address):
+    Bar(_addr).bar(456)
+"""
+    c1 = get_contract(contract_1)
+    c2 = get_contract(contract_2)
+    assert_tx_failed(lambda: c2.foo(c1.address))
+
+
+def test_returndatasize_empty(get_contract, assert_tx_failed):
+    contract_1 = """
+@external
+def bar(a: int128):
+    pass
+"""
+    contract_2 = """
+interface Bar:
+    def bar(a: int128) -> int128: view
+
+@external
+def foo(_addr: address) -> int128:
+    return Bar(_addr).bar(456)
+"""
+    c1 = get_contract(contract_1)
+    c2 = get_contract(contract_2)
+    assert_tx_failed(lambda: c2.foo(c1.address))
+
+
+def test_returndatasize_too_long(get_contract, assert_tx_failed):
+    contract_1 = """
+@external
+def bar(a: int128) -> (int128, int128):
+    return a, 789
+"""
+    contract_2 = """
+interface Bar:
+    def bar(a: int128) -> int128: view
+
+@external
+def foo(_addr: address) -> int128:
+    return Bar(_addr).bar(456)
```

Comment:
Does this mean that the following _also_ works?

```python
@external
def baz(_addr: address) -> (int128, int128):
    return Bar(_addr).bar(456)
```

Reply:
Should probably also check this:

```python
@external
def baz(_addr: address) -> int128:
    a, _ = Bar(_addr).bar(456)
    return a
```

Reply:
The first would work, yes.  And the 2nd won't work because type checking will flag it as expecting the wrong number of arguments.

Reply:
Oh yeah, the interface `Bar` doesn't match the return type of bar in the other contract. So shouldn't both fail to compile?

Reply:
They are seperate contracts - they can't see each other when compiling.  The check here is that when B calls A, but A returns _more_ calldata than expected, we don't raise.  This is consistent with how Solidity handles it.

Reply:
Yeah but in the first example the `Bar` interface specifies it returns an `int128`, but then in `baz` you're saying you expect it to return a `(int128, int128)`, which directly conflicts with the interface. Doesn't have anything to do with the fact the actual contract returns `(int128, int128)`, this should fail a compile-time check.